### PR TITLE
fix: put a checklist tick back when the server refuses to save it

### LINKED ISSUE #204: Checklist tick stays checked locally when the description update fails
Follow-up from Copilot's review on #202 (checklists inside descriptions, #200, shipped in 1.17.0 build 69).

## Problem

`toggleChecklistItem` in both detail views writes the toggled description into the view's local state first, then calls `AppState.updateTask` with a description-only request:

- `mDone/Views/Tasks/TaskDetailSheet.swift` (iOS)
- `mDone/Views/Mac/MacTaskDetailView.swift` (macOS)

`updateTask` catches server errors, shows the error banner via `handleError`, and returns without reporting failure. So if the POST is rejected, the checkbox and the "n of m" summary stay in the ticked state while Vikunja still has the old description. Cancel then closes the sheet with the user believing the tick was saved.

The offline case is fine: `queueOfflineEdit` queues the edit and the local state is correct. On macOS, `checklistSaveInFlight` also stays set after a failure; harmless, since it only suppresses a reload whose description matches the failed one exactly, but it should be cleared on the same path.

## Fix

Have `updateTask` report success (return `Bool`, or a small result enum for saved / queued / failed), and in both `toggleChecklistItem` implementations restore the previous description (and clear `checklistSaveInFlight` on macOS) when it fails. Add a unit test for the failure path.

Deliberately not fixed in #201/#202 because build 69 with that code was already on TestFlight and the edge case did not justify a build 70.

### PR DESCRIPTION
**Pull request by marco308**
Fixes #204, the finding Copilot left open on #202.

A checklist tap in the task detail showed the tick before saving it, and `AppState.updateTask` swallowed server errors, so a refused POST left the box ticked while Vikunja still had it unticked. Cancel then closed the sheet with the user believing the tick was saved.

**Change**

- `AppState.updateTask` now returns a discardable `Bool`: `true` when the edit reached the server or was queued for offline replay, `false` when nothing was saved. Every existing caller ignores it, so nothing else changes behaviour.
- Both `toggleChecklistItem` implementations (iOS `TaskDetailSheet`, Mac `MacTaskDetailView`) restore the previous description when the save fails, unless the description has moved on since. The Mac view also drops its `checklistSaveInFlight` marker so the next outside refresh reloads the form as usual.
- The usual error banner still shows, so the user sees why the tick came back.

**Tests**

- `AppStateTests`: `updateTask` returns `true` on a 200 and `false` on a 400, and the local task keeps the server's copy on failure.
- `OfflineEditQueueTests`: a queued offline edit returns `true`, so the tick stays.
- Ran `AppStateTests`, `OfflineEditQueueTests` and `DescriptionChecklistTests` on the iPhone 17 simulator: 84 tests, 0 failures. macOS app builds.

Not in build 69, which already went to TestFlight; the next cut should carry it together with `261543c`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: mDone/Views/Mac/MacTaskDetailView.swift [hunk 1]
```diff
@@ -254,15 +254,28 @@ struct MacTaskDetailView: View {
...
+            // The tick was shown before it was saved. If the server refused
+            // it, put the box back the way Vikunja still has it, unless the
+            // description has moved on since, and forget the in-flight
+            // marker so the next outside refresh reloads the form (#204).
+            if !saved {
+                if checklistSaveInFlight == updated {
+                    checklistSaveInFlight = nil
+                }
+                if descriptionText == updated {
+                    descriptionText = previous
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Mac/MacTaskDetailView.swift]: Fixed in 46f705a. The Mac rollback now restores the description from the task in AppState (which is what the server still has) rather than the text this tap started from, so two quick failing taps end on the original text. The "description has moved on" guard stays so an in-flight later edit is not clobbered.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: mDone/Views/Tasks/TaskDetailSheet.swift [hunk 2]
```diff
@@ -307,14 +307,21 @@ struct TaskDetailSheet: View {
...
         let committedEstimate = (appState.tasks.first(where: { $0.id == task.id }) ?? task).estimatedSeconds
         let composed = EstimateMarker.apply(committedEstimate, to: updated) ?? ""
         Task { @MainActor in
-            await appState.updateTask(id: task.id, request: TaskUpdateRequest(description: composed))
+            let saved = await appState.updateTask(id: task.id, request: TaskUpdateRequest(description: composed))
+            // The tick was shown before it was saved. If the server refused
+            // it, put the box back the way Vikunja still has it, unless the
+            // description has moved on since (#204).
+            if !saved, description == updated {
+                description = previous
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/Views/Tasks/TaskDetailSheet.swift]: Fixed in 46f705a. Same change as on Mac: the iOS rollback restores the description from the task in AppState instead of the tap's `previous`, so after two quick failing taps the view shows the original text the server still has.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: mDone/App/AppState.swift [hunk 3]
```diff
@@ -1287,15 +1287,21 @@ final class AppState {
         }
     }
 
+    /// Returns `true` when the edit reached the server or was queued for
+    /// replay while offline, `false` when nothing was saved (the request
+    /// failed, or another update for the same task was already running).
```
[the remark this answers is not in the document]
Comment (marco308) [reply mDone/App/AppState.swift]: Fixed in 46f705a. The doc now says `false` means the request failed or the calling task was cancelled, and that an update that finds the task busy waits its turn rather than failing.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
